### DISCUSSION TITLE
Need for rendering all predicted frames trainable?

### DISCUSSION THREAD

**Discussion by chrisch12138** [talmolab/sleap#1709] [Help!] [answered]
Hi guys,

I have videos of groups of mice interacting and I am going through prediction assisted labeling process. I'm just wondering if it would be beneficial to render the predicted instances that are very correct as a user-labeled instance for the model to train? Or should I just correct the instances with nodes that are wrongly predicted?
I wonder if in a certain frame only a portion of animals are user-labeled, would that affect the training and validation process (as it may seem each frame is seen as an entity?)

Thanks!

**Comment by roomrys** [comment, marked as answer]
Hi @chrisch12138,

> I'm just wondering if it would be beneficial to render the predicted instances that are very correct as a user-labeled instance for the model to train? Or should I just correct the instances with nodes that are wrongly predicted?

We usually advise people to label an even spread of accurate to inaccurate predictions. Turning the good predictions into labels will help reinforce that those predictions remain so in the next model iteration. Correcting the inaccurate predictions will help teach the model what was expected. So there is benefit from doing both. However, if you already have plenty of labels of a particular pose which the model performs well on, then I would focus on other poses that the model still has trouble on.

>I wonder if in a certain frame only a portion of animals are user-labeled, would that affect the training and validation process (as it may seem each frame is seen as an entity?)

Although we don't make use of any explicit negative examples, leaving keypoints unlabeled that are clearly visible only leads to confusion on whether or not there is anything different about that keypoint that warrants it being unlabeled. Furthermore, examples for training are passed in on a frame-by-frame basis, so always fully label a frame or do not label it at all. If any keypoints are occluded, you can right-click the node to mark them as not visible.

Thanks,
Liezl

**Reply by chrisch12138** [reply]
Thank you!
